### ISSUE TITLE
Stubs: DS Vector map returns Invalid argument

### ISSUE THREAD

**Issue by jarstelfox** [vimeo/psalm#2786]
https://psalm.dev/r/61fdcc8276

Psalm dev actually seems to throw an internal exception on this case.

On psalm `3.8.3@389af1bfc739bfdff3f9e3dc7bd6499aee51a831`: 
```php
ERROR: InvalidArgument - test.php:7:9 - Argument 1 of Ds\Vector::map expects callable(int):TNewValue:fn-ds\sequence::map as mixed, Closure(int):string provided
$v->map($identity
```

**Comment by muglug** [comment]
I've fixed the exception (by installing `ext-ds` on the server) so the example gives the right error now.

**Comment by muglug** [comment]
Fixed in 7639e179c4a2da2c28ab6a2a0e6112f2d3111050 cc @enumag

**Comment by enumag** [comment]
@muglug Thanks for the ping. I ported the fix to phpstan too: https://github.com/phpstan/phpstan-src/pull/139
